lib/msal-angular/src/msal.service.ts

```diff
@@ -189,12 +160,24 @@ export class MsalService extends UserAgentApplication {
         }
 
         const protectedResourceMap = frameworkProtectedResourceMap && frameworkProtectedResourceMap.size ? frameworkProtectedResourceMap : new Map(this.msalAngularConfig.protectedResourceMap);
-
-        // process all protected resources and send the matched one
-        const keyForEndpoint = Array.from(protectedResourceMap.keys()).find(key => endpoint.indexOf(key) > -1);
-        if (keyForEndpoint) {
-            return protectedResourceMap.get(keyForEndpoint);
-        }
+        
+        const protectedResourcesArray = Array.from(protectedResourceMap.keys());
+        const keyMatchesEndpointArray = protectedResourcesArray.filter(key => {
+            const minimatch = new Minimatch(key);
+            return minimatch.match(endpoint);
```

Comment:
```suggestion
            return minimatch.match(endpoint) || endpoint.indexOf(key) > -1;
```

Reply:
You can test this with a test case of `http://localhost:3000` in the protected resource map and `http://localhost:3000/api` as the endpoint (which should work today).